plugins/groovy/test/org/jetbrains/plugins/groovy/inspections/GrChangeToOperatorTest.groovy

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2000-2016 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jetbrains.plugins.groovy.inspections
+
+import com.intellij.testFramework.fixtures.LightCodeInsightFixtureTestCase
+import org.intellij.lang.annotations.Language
+import org.jetbrains.plugins.groovy.codeInspection.changeToOperator.ChangeToOperatorInspection
+
+import java.util.regex.Pattern
+
+import static org.jetbrains.plugins.groovy.GroovyFileType.GROOVY_FILE_TYPE
+import static org.jetbrains.plugins.groovy.util.TestUtils.CARET_MARKER
+
+public class GrChangeToOperatorTest extends LightCodeInsightFixtureTestCase {
+  def inspection = new ChangeToOperatorInspection()
+
+  void testSimpleUnaryExpression() {
+    assertValid(/a.bitwiseNegate()/, /~a/)
+    assertValid(/a.negative()/, /-a/)
+    assertValid(/a.positive()/, /+a/)
+    assertValid(/a.call()/, /a()/)
+    assertValid(/a.next()/, /++a/)
+    assertValid(/a.previous()/, /--a/)
+  }
+
+  void testNegatableUnaryExpression() {
+    assertValid(/a.asBoolean()/, /!!a/)
+    assertValid(/!a.asBoolean()/, /!a/)
+
+    assertValid(/if (${_}a.asBoolean()${_});/, /a/)
+    assertValid(/if (${_}!a.asBoolean()${_});/, /!a/)
+  }
+
+  // TODO  
```

Comment:
@dovchinnikov, as I stated in a previous comment also
> `methodCallExpression.reolveMethod` isn't working for complex scenarios (e.g. [1,2,3].isCase(0))

I guess a separate issue should be opened for this.